File: javalin6/javalin-html-forms-example/src/main/java/app/JavalinHtmlFormsExampleApp.java
package app;

import io.javalin.Javalin;
import io.javalin.http.staticfiles.Location;
import io.javalin.util.FileUtil;
import java.util.HashMap;
import java.util.Map;

public class JavalinHtmlFormsExampleApp {

    private static final Map<String, String> reservations = new HashMap<>() {{
        put("saturday", "No reservation");
        put("sunday", "No reservation");
    }};

    public static void main(String[] args) {

        Javalin app = Javalin.create(config -> {
            config.staticFiles.add("/public", Location.CLASSPATH);
        });

        app.post("/make-reservation", ctx -> {
            reservations.put(ctx.formParam("day"), ctx.formParam("time"));
            ctx.html("Your reservation has been saved");
        });

        app.get("/check-reservation", ctx -> {
            ctx.html(reservations.get(ctx.queryParam("day")));
        });

        app.post("/upload-example", ctx -> {
            ctx.uploadedFiles("files").forEach(file -> {
                FileUtil.streamToFile(file.content(), "upload/" + file.filename());
            });
            ctx.html("Upload successful");
        });

        app.start();

    }

}

File: javalin7/javalin-html-forms-example/src/main/kotlin/JavalinHtmlFormsExampleApp.kt
import io.javalin.Javalin
import io.javalin.http.staticfiles.Location
import io.javalin.util.FileUtil

private val reservations = mutableMapOf<String?, String?>(
    "saturday" to "No reservation",
    "sunday" to "No reservation"
)

fun main() {
    val app = Javalin.create { config ->
        config.staticFiles.add("/public", Location.CLASSPATH)
        config.routes.post("/make-reservation") { ctx ->
            reservations[ctx.formParam("day")] = ctx.formParam("time")
            ctx.html("Your reservation has been saved")
        }
        config.routes.get("/check-reservation") { ctx ->
            ctx.html(reservations[ctx.queryParam("day")]!!)
        }
        config.routes.post("/upload-example") { ctx ->
            ctx.uploadedFiles("files").forEach {
                FileUtil.streamToFile(it.content, "upload/${it.filename}")
            }
            ctx.html("Upload successful")
        }
    }.start(7070)
}
